cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -57,6 +58,35 @@ class SimpleArray; // forward declaration
 namespace detail
 {
 
+template <typename T>
+concept IntegralType = std::is_integral_v<T>;
```

Comment:
Concepts should be made in `modmesh` namespace rather than `modmesh::detail`.  Thus we should be careful creating and naming them.

Reply:
Moved all concepts outside `detail`